src/software/SfM/main_openMVG2openMVS.cpp

```diff
@@ -28,219 +28,346 @@ using namespace openMVG::sfm;
+    const std::string& sOutDir
+)
+{
+    if (!stlplus::is_folder(sOutDir))
     {
-      std::cout << "Cannot read the corresponding image: " << srcImage << std::endl;
-      return EXIT_FAILURE;
+        stlplus::folder_create(sOutDir);
+        if (!stlplus::is_folder(sOutDir))
+        {
+            std::cerr << "Cannot access to one of the desired output directory" << std::endl;
+            return false;
+        }
     }
-    if (sfm_data.IsPoseAndIntrinsicDefined(view.second.get()) && stlplus::is_file(srcImage))
+
+    std::map<const openMVG::cameras::IntrinsicBase*, std::vector<Image<double>*>> map_cam_remap;
+
+    bool ret = true;
+
+    for (const auto& view : sfm_data.GetViews())
     {
-      MVS::Interface::Platform::Pose pose;
-      image.poseID = platform.poses.size();
-      const openMVG::geometry::Pose3 poseMVG(sfm_data.GetPoseOrDie(view.second.get()));
-      pose.R = poseMVG.rotation();
-      pose.C = poseMVG.center();
-      // export undistorted images
-      const openMVG::cameras::IntrinsicBase * cam = sfm_data.GetIntrinsics().at(view.second->id_intrinsic).get();
-      if (cam->have_disto())
-      {
-        // undistort image and save it
-        Image<openMVG::image::RGBColor> imageRGB, imageRGB_ud;
-        ReadImage(srcImage.c_str(), &imageRGB);
-        UndistortImage(imageRGB, cam, imageRGB_ud, BLACK);
-        WriteImage(outputImage.c_str(), imageRGB_ud);
-        // undistort mask and save it
-        auto maskFileName = srcImage + ".mask.png";
-        if (stlplus::is_file(maskFileName)) {
-          Image<openMVG::image::RGBColor> maskRGB, maskRGB_ud;
-          ReadImage(maskFileName.c_str(), &maskRGB);
-          UndistortImage(maskRGB, cam, maskRGB_ud, BLACK);
-          WriteImage((outputImage + ".mask.png").c_str(), maskRGB_ud);
+        const std::string srcImage = stlplus::create_filespec(sfm_data.s_root_path, view.second->s_Img_path);
+
+        const std::string outputImage = stlplus::create_filespec(sOutDir, view.second->s_Img_path);
+
+        if (!stlplus::is_file(srcImage))
+        {
+            std::cout << "Cannot read the corresponding image: " << srcImage << std::endl;
+            ret = false;
+            break;
         }
-        // undistort depth and save it
-        auto depthFileBasename = view.second->s_Img_path;
-        auto pos = depthFileBasename.find("rgb");
-        if (pos != std::string::npos) {
-            depthFileBasename.replace(pos, 3, "depth");
+        if (sfm_data.IsPoseAndIntrinsicDefined(view.second.get()))
+        {
+            // export undistorted images
+            const openMVG::cameras::IntrinsicBase* cam = sfm_data.GetIntrinsics().at(view.second->id_intrinsic).get();
+
+            if (cam->have_disto())
+            {
+                Image<double> *map_row = NULL;
+                Image<double> *map_col = NULL;
+
+                auto iterMapCam = map_cam_remap.find(cam);
+                if (map_cam_remap.end() != iterMapCam)
```

Comment:
Change following code like this:

if (map_cam_remap.end() != iterMapcam) {
  CalcCameraDistortionMap(cam, map_cam_remap[cam][0], map_cam_remap[cam][1]);
}

auto &map_row = map_cam_remap[cam][0];
auto &map_col = map_cam_remap[cam][1];